Develop

savvy-sam opened the pull request.

savvy-sam committed:
[chore] add bootstrap css to yummy

savvy-sam committed:
[chore] add bootstrap js to Yummy

savvy-sam committed:
[chore] add wireframes and ORM desgns to yummy

savvy-sam committed:
[feature] create user login form

savvy-sam committed:
[chore] add css styling to yummy

savvy-sam committed:
[feuture] add registration form to yummy

savvy-sam committed:
[feuture] add the recipe index form to yummy

paulupendo reviewed (commented):
Good job. Kindly check for typos in your readme and ensure correct spelling when writing your commit messages next time. You can improve the quality of your Readme by checking out the readme assessment guidelines on challenge doc e.g You can add screenshots of your app, use markdown to style your Readme.

savvy-sam committed:
install pipenev to for mananging packages. install virtualenv and
virtualenv wrapper to create virtual environments. Create a virtual
environment and switch to it. Install requirements.txt for specifying
which packages the app will use. Finnaly add gitignore file to the base
of the application. This will prevent pushing of some items to the
remote
repository

savvy-sam committed:
[chore] mapped out the irectory structure of the app

savvy-sam committed:
[chore]initialized a flask app

savvy-sam committed:
[feature] Add index page to the app

savvy-sam committed:
[feature]intergrate login page to the app

savvy-sam committed:
[feature] intergrate sign up form into the app

savvy-sam committed:
[feature] add the recipes index form to the app

savvy-sam committed:
[chore] add unit tests to the flask app

savvy-sam committed:
Update README.md

[chore] update README to meet the required standards

savvy-sam committed:
[chore] added tests to the yummy app

savvy-sam committed:
Merge branch 'develop' of https://github.com/savvy-sam/yummy into develop
[chore] update the develop branch from the remote branch

savvy-sam committed:
[chore] created the  static folder from where  css will be served

savvy-sam committed:
Update README.md

[chore] added travisCI badge to readme file

savvy-sam committed:
Update README.md

savvy-sam committed:
Update README.md

[chore] added licence badge to readme

savvy-sam committed:
Update README.md

savvy-sam committed:
[chore] added gitignore

savvy-sam committed:
Merge branch 'develop' of https://github.com/savvy-sam/yummy into develop

savvy-sam committed:
[chore] remove runtime files from yummy

savvy-sam committed:
[chore] delete .cache file from yummy

savvy-sam committed:
[chore] update readme

savvy-sam committed:
[chore] edit readme file

savvy-sam committed:
[chore] add badges to yummy

savvy-sam committed:
[chore] add .travis file to yummy

savvy-sam committed:
[chore] populate travis.yml

savvy-sam committed:
[chore] modified travis.yml file

savvy-sam committed:
[test] added a travis badge to readme

savvy-sam committed:
[test] test if the travis badge on readme is working

savvy-sam committed:
[edit] clean up the readme file to remove non-working badges

savvy-sam committed:
[edit] renove unneeded elements from readme

savvy-sam committed:
[feature] add a register form to collect data from the user.Then add a
model.py file to define the user class. Finally add a route to define
how to store the data from the register form.

savvy-sam committed:
[feature] add the Recipes class to to models.py . After that, add CRUD
methods Create and Update to recipes. Create a form to take imput from
the user

savvy-sam committed:
[feature] add update method for the User class. The update method uses the same form used fo create

savvy-sam committed:
[feature] add a route for displaying all users and recipes

savvy-sam committed:
[tests] add test to test whether User and Recipe classes are creating
succesfully. Edit comments and docstrings to make your code
understandable

savvy-sam committed:
[edits] edited files to meet PEP standards

savvy-sam committed:
[edit] edit files to clear out errors

savvy-sam committed:
[update] updated .travis.yml to cover travis builds

savvy-sam committed:
[edit] edited the Readme file to add working travis and coveralls badges

savvy-sam committed:
[feature] declare the user variable as global variable so that it can
be accessed by all methods. Previously, user was defined as
a local variable in the create method. It was therefore not available to
show and index methods

savvy-sam committed:
[feature] Add login functionality to the yummy app. this involves
creating a form to collect user data then creating a login method to
save the user id to sessions

savvy-sam committed:
[edit] modified app_test.py to use uniform identation

savvy-sam committed:
[environment] create a virtual environment and switch to it. install
requirement.text to store your packages.Edit .travis.yml to use the
correct python version

savvy-sam committed:
[environment] installed flask and WTForms in virtualenv

savvy-sam committed:
[builds] edited .trvis.yml to fix failing travis builds

savvy-sam committed:
[builds] edited .trvis.yml to fix failing travis builds

savvy-sam committed:
[builds] edited .trvis.yml to fix failing travis builds

savvy-sam committed:
[edit] added an updated coveralls badge to readme

savvy-sam committed:
[tests] added .coveralls.yml to track tests coverage

savvy-sam committed:
[feature] add jinja code to recipe_index.html to render a list of all recipes created

savvy-sam committed:
[feature] add a html template to show the title and content of a recipe.
also add a route that renders the show recipe template

savvy-sam committed:
[feature] add delete functionality to recipes. The delete functionality
will work by removing a recipe from the RECIPES_INDEX list

savvy-sam committed:
[edits] Edit the names of routes to follow a consistent naming .Remove
routes that are no longer in use.

savvy-sam committed:
[edit] remove tests for deleted routes

savvy-sam committed:
[tests] add coveralls for test coverage reporting

savvy-sam committed:
[edit] remove packages that are not needed from requirement.txt

savvy-sam committed:
[tests]Add tests to check whether http method GET is succesful in all
routes

savvy-sam committed:
[tests] add tests to test User and Recipe objects

savvy-sam committed:
[tests] add tests to inspect POST request for User, Recipe methods.
modify the RecipeForm class to include validators for presence, i.e to
ensure that a particular field must be filled

savvy-sam committed:
[feature] Add Procfile to the repository. This file is needed for
succesful deployment on heroku. The file specifies the scripts that will
run when the app is deployed. It also specifies the number of dynos on
which the application will run on heroku. Modify run.py to specify the
port on which the app will run

savvy-sam committed:
[rename rename requirement.txt to reqirements.txt. Heroku builds require
a file requirements.txt on the base of the repository

savvy-sam committed:
[fixes]updated appdir to fix failing heroku builds. imported os module
to fix undefined name error.

savvy-sam committed:
[refactor] display recipe id in the recipe index page. Change the User
and Recipe models to generate integer ids. Edit the delete route to take
variables

savvy-sam committed:
[refactor] edit the views.py to create a logout method. Also edit the
update and the delete method to take the recipe id as a parameter. Add a
user_id attribute to recipes so that recipes can be associated with a
particular user

savvy-sam committed:
[refactor] edited the user update route to take parameters

savvy-sam committed:
[feature] add a method that isolates all recipes with a particular user_id into a list

savvy-sam committed:
[deletes] remove installations from your repo.  Virtual environment
variables shouldnt be pushed to remote repo.

savvy-sam committed:
[badges] add markdown badges to readme

savvy-sam committed:
[linting] edit model.py and views.py to PEP8 standards. Also specify the
tests coverage in .travis.yml

savvy-sam committed:
[feauture] add more fields to recipe to make it comprehensive

savvy-sam committed:
[features] make a form and define a method for creating categories.
Define a route for returning all the recipes contained in a particular
category

savvy-sam committed:
[feature] added a method to delete a categories

savvy-sam committed:
[style] edited templates to make them visually appealing

savvy-sam committed:
[templates] edit templates to use templates inheritance. Also add
buttons and navigation links

savvy-sam committed:
[Tests] edited tests to fix failing tests

savvy-sam committed:
[styling] edit the front end of the recipe to make it more appealing

savvy-sam committed:
[style] edit the UI templates to make them more appealing and add links
for easier navigation

savvy-sam committed:
[validations] add view decorators to all functions that require login

savvy-sam committed:
[validations] Add a method that validates whether the user email exists.
user emails should be unique

savvy-sam committed:
[documentation] add more details to readme file to include added
changes. Also add screenshots of  yummy app to readme

savvy-sam committed:
[templates] add a link to update user details to base.html

savvy-sam committed:
[UI templates] create and style the page to show a recipe

savvy-sam committed:
[tests] add tests to factor in decorators

savvy-sam committed:
[tests] added tests to category methods

savvy-sam committed:
[tests] add more tests to increase coverage

savvy-sam committed:
[Tests builds] adjust travis.yml to cover models

savvy-sam committed:
[Tests builds] adjust travis.yml to resolve failing builds

savvy-sam commented:
couldn't make it to the fellowship. It is devastating but i will try next
time


On Fri, Nov 3, 2017 at 11:40 AM, Coveralls <notifications@github.com> wrote:

> [image: Coverage Status] <https://coveralls.io/builds/14021391>
>
> Changes Unknown when pulling *94e7a8b
> <https://github.com/savvy-sam/yummy/commit/94e7a8bf2989694c1359fcda5c1f44efaab9cf6e>
> on develop* into ** on master**.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/savvy-sam/yummy/pull/1#issuecomment-341646222>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AfRiolbsHgl6wfqPfRiazj_f8pHnvu5pks5sytFjgaJpZM4P9hJp>
> .
>
